flink-java/src/main/java/org/apache/flink/api/java/utils/PythonDependencyManager.java

```diff
@@ -0,0 +1,282 @@
+		}
+	}
+
+	public void configure(ReadableConfig config) {
+		config.getOptional(PythonOptions.PYTHON_FILES).ifPresent(pyFiles -> {
+			for (String filePath : pyFiles.split(FILE_DELIMITER)) {
+				addPythonFile(filePath);
+			}
+		});
+
+		config.getOptional(PythonOptions.PYTHON_REQUIREMENTS).ifPresent(pyRequirements -> {
+			if (pyRequirements.contains(PARAM_DELIMITER)) {
+				String[] requirementFileAndCache = pyRequirements.split(PARAM_DELIMITER, 2);
+				setPythonRequirements(requirementFileAndCache[0], requirementFileAndCache[1]);
+			} else {
+				setPythonRequirements(pyRequirements);
+			}
+		});
+
+		config.getOptional(PythonOptions.PYTHON_ARCHIVES).ifPresent(pyArchives -> {
+			for (String archive : pyArchives.split(FILE_DELIMITER)) {
+				String archivePath;
+				String targetDir;
+				if (archive.contains(PARAM_DELIMITER)) {
+					String[] filePathAndTargetDir = archive.split(PARAM_DELIMITER, 2);
+					archivePath = filePathAndTargetDir[0];
+					targetDir = filePathAndTargetDir[1];
+				} else {
+					archivePath = archive;
+					targetDir = null;
+				}
+				addPythonArchive(archivePath, targetDir);
+			}
+		});
+
+		config.getOptional(PYTHON_EXECUTABLE).ifPresent(executable -> {
+			this.config.set(PYTHON_EXECUTABLE, executable);
+		});
+
+		config.getOptional(PYTHON_CLIENT_EXECUTABLE).ifPresent(executable -> {
+			this.config.set(PYTHON_CLIENT_EXECUTABLE, executable);
+		});
+	}
+
+	private String generateUniqueFileKey(String configKey) {
+		counter += 1;
+		return String.format("%s_%d_%s", configKey, counter, UUID.randomUUID());
+	}
+
+	private List<String> getRegisteredFile(String keyPrefix, String path) {
+		List<String> result = new ArrayList<>();
+		for (Tuple2<String, DistributedCache.DistributedCacheEntry> t : cachedFiles) {
+			if (t.f0.startsWith(keyPrefix)) {
+				if (t.f1.filePath.equals(path)) {
+					result.add(t.f0);
+				}
+			}
+		}
+		return result;
+	}
+
+	private boolean fileAlreadyRegistered(String path) {
+		return !getRegisteredFile(PYTHON_FILE_PREFIX, path).isEmpty();
+	}
+
+	private boolean archiveAlreadyRegistered(String archivePath, String targetDir) {
+		List<String> registeredFileKey = getRegisteredFile(PYTHON_ARCHIVE_PREFIX, archivePath);
+		if (!registeredFileKey.isEmpty()) {
+			for (String fileKey : registeredFileKey) {
+				if (targetDir.equals(config.get(PYTHON_ARCHIVES).get(fileKey))) {
+					// current archive is registered, skip it.
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+	private void removeCachedFileIfExists(String fileKey) {
+		for (Tuple2<String, DistributedCache.DistributedCacheEntry> t : cachedFiles) {
+			if (fileKey.equals(t.f0)) {
+				cachedFiles.remove(t);
+				break;
+			}
+		}
+	}
+
+	private void registerCachedFile(String filePath, String fileKey) {
+		cachedFiles.add(new Tuple2<>(fileKey, new DistributedCache.DistributedCacheEntry(filePath, false)));
+	}
+
+	public static void synchronizePythonDependencyMetadata(Configuration from, Configuration to) {
```

Comment:
What about add a method such as `getPythonConfiguration` and then the caller could use `from.addAll(getPythonConfiguration(to))` to copy the Python related configurations.